src/libstd/collections/hash/table.rs

```diff
@@ -34,6 +34,37 @@ type HashUint = usize;
 
 const EMPTY_BUCKET: HashUint = 0;
 
+/// Special `Unique<T>` that uses the lower bit of the pointer to expose
+/// a boolean tag. Can only be used with T's with alignment > 1.
+struct TaggedUnique<T>(Unique<T>);
+
+impl<T> TaggedUnique<T> {
+    unsafe fn new(ptr: *mut T) -> Self {
+        assert!(mem::align_of::<T>() > 1);
```

Comment:
1 is the smallest alignment possible, so this assertion is always triggered, no?